### PR TITLE
Bootstrap/Scan/Watch `ParentNode`

### PR DESCRIPTION
**Pull request by dahlbyk**
Another patch for Turbo: streams are returned inside a `template` which you can't query into directly; you need to use their [`content` property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLTemplateElement/content), which is a `DocumentFragment`. Fortunately the DOM TS bindings provide a suitable common base interface: `ParentNode`.

Changing to `ParentNode` required a few adjustments that ultimately yield more precise type checks.

### CODE REVIEW

File: dist/aspnet-validation.js [hunk 1]
```diff
@@ -893,10 +893,10 @@ var ValidationService = /** @class */ (function () {
      * Scans the entire document for input elements to be validated.
      */
     ValidationService.prototype.scanInputs = function (root) {
-        var inputs = Array.from(root.querySelectorAll('[data-val="true"]'));
+        var inputs = Array.from(root.querySelectorAll('input[data-val="true"]'));
```
Comment (haacked) [inline dist/aspnet-validation.js:896]: What about `textarea` and `select`. Can we really rely on `input` here?

File: src/index.ts [hunk 2]
```diff
@@ -884,17 +884,17 @@ export class ValidationService {
     /**
      * Scans the entire document for input elements to be validated.
      */
-    private scanInputs(root: HTMLElement) {
-        let inputs = Array.from(root.querySelectorAll<HTMLElement>('[data-val="true"]'));
+    private scanInputs(root: ParentNode) {
+        let inputs = Array.from(root.querySelectorAll<HTMLInputElement>('input[data-val="true"]'));
```
Comment (haacked) [inline src/index.ts:888]: Whoops, I commented on the wrong file. I'm not sure about this change. Do we scan on `textarea` and `select` elsewhere?

Comment (dahlbyk) [reply src/index.ts:888]: Good catch. Was aligning the selector with the `HTMLInputElement` a few lines down, but we should probably keep it more generic.

Comment (dahlbyk) [reply src/index.ts]: Alright, took a stab at replacing `HTMLInputElement` with a new `ValidatableElement`.

File: src/index.ts [hunk 3]
```diff
@@ -83,13 +108,13 @@ function getRelativeFormElement(element: HTMLInputElement, selector: string) {
         // Form.Password
         let relativeElementName = objectName + '.' + realSelector;
         let relativeElement = document.getElementsByName(relativeElementName)[0];
```
Comment (dahlbyk) [inline src/index.ts:110]: Not new here, but I wonder if this should be constrained to `element.form` instead of `document`?

Comment (haacked) [reply src/index.ts:110]: Possibly. Seems out of scope for this PR though. I'd want to see a demo in `Index.cshtml` of the problem this causes.

**Comment by haacked** [comment]
BTW, I did a force push (sorry), so you'll probably want to do a reset against origin on your local machine.

File: src/index.ts [hunk 4]
```diff
@@ -26,6 +26,31 @@ const nullLogger = new (class implements Logger {
...
+
+const validatableElementTypes = ['input', 'select', 'textarea'];
+
+/**
+ * Generates a selector to match validatable elements (`input`, `select`, `textarea`).
+ * @param selector An optional selector to apply to the valid input types, e.g. `[data-val="true"]`.
+ * @returns The validatable elements.
+ */
+const validatableSelector = (selector?: string) =>
+    validatableElementTypes.map(t => `${t}${selector || ''}`).join(',');
```
[the remark this answers is not in the document]
Comment (dahlbyk) [reply src/index.ts:52]: Oops, yeah I messed up the callsite. 👇

**Review by dahlbyk — COMMENTED** [review]
Great catch

**Comment by dahlbyk** [comment]
> I'll probably rename the variable for clarity.

Review again?

**Pull request merged**